xarray/tests/test_dataset.py

```diff
@@ -3495,6 +3495,31 @@ def test_setitem_str_dtype(self, dtype):
 
         assert np.issubdtype(ds.x.dtype, dtype)
 
+    def test_setitem_using_list(self):
+
+        # assign a list of variables
+        var1 = Variable(["dim1"], np.random.randn(8))
+        var2 = Variable(["dim1"], np.random.randn(8))
+        actual = create_test_data()
+        expected = actual.copy()
+        expected["A"] = var1
+        expected["B"] = var2
+        actual[["A", "B"]] = [var1, var2]
+        assert_identical(actual, expected)
+        # assign a list of numpy arrays
+        actual[["C", "D"]] = [var1.values, var2.values]
+        expected[["C", "D"]] = expected[["A", "B"]]
+        assert_identical(actual, expected)
+
+        with pytest.raises(ValueError, match=r"Different lengths"):
+            actual[["A", "B"]] = [var1]
+
+        with pytest.raises(ValueError, match=r"Empty list of variables"):
+            actual[[]] = [var1]
+
+        with pytest.raises(ValueError, match=r"assign single DataArray"):
+            actual[["A", "B"]] = xr.DataArray([1, 2])
```

Comment:
Could we put this in a separate, parametrized test? I imagine something along these lines:

```diff
diff --git a/xarray/tests/test_dataset.py b/xarray/tests/test_dataset.py
index d00e6583..19849ffc 100644
--- a/xarray/tests/test_dataset.py
+++ b/xarray/tests/test_dataset.py
@@ -3511,14 +3511,22 @@ class TestDataset:
         expected[["C", "D"]] = expected[["A", "B"]]
         assert_identical(actual, expected)
 
-        with pytest.raises(ValueError, match=r"Different lengths"):
-            actual[["A", "B"]] = [var1]
-
-        with pytest.raises(ValueError, match=r"Empty list of variables"):
-            actual[[]] = [var1]
-
-        with pytest.raises(ValueError, match=r"assign single DataArray"):
-            actual[["A", "B"]] = xr.DataArray([1, 2])
+    @pytest.mark.parametrize(
+        "var_list, data, error_regex",
+        [
+            (
+                ["A", "B"],
+                [Variable(["dim1"], np.random.randn(8))],
+                r"Different lengths",
+            ),
+            ([], [Variable(["dim1"], np.random.randn(8))], r"Empty list of variables"),
+            (["A", "B"], xr.DataArray([1, 2]), r"assign single DataArray"),
+        ],
+    )
+    def test_setitem_using_list_errors(self, var_list, data, error_regex):
+        actual = create_test_data()
+        with pytest.raises(ValueError, match=error_regex):
+            actual[var_list] = data
 
     def test_assign(self):
         ds = Dataset()
```